btc-lsp/src/BtcLsp/Thread/LnChanWatcher.hs

```diff
@@ -0,0 +1,25 @@
+module BtcLsp.Thread.LnChanWatcher
+  (watchChannelEvents, forkThread)
+where
+
+import BtcLsp.Import hiding (takeMVar, putMVar, newEmptyMVar)
+import UnliftIO.Concurrent (ThreadId, forkFinally, putMVar)
+import BtcLsp.Import (newEmptyMVar)
+import qualified LndClient.RPC.Silent as Lnd
+import BtcLsp.Storage.Model.LnChan (persistChannelUpdates)
+import LndClient
+
+forkThread :: MonadUnliftIO m => m () -> m (ThreadId, MVar ())
+forkThread proc = do
+    handle <- newEmptyMVar
+    tid <- forkFinally proc (\_ -> putMVar handle ())
```

Comment:
Let's use Async to supervise threads nicely and uniformly without implementing low-level stuff in thread itself. 

Example from lsp:
https://github.com/coingaming/src/blob/13285468612f0c4b709c5489279dd189896d3143/btc-lsp/src/BtcLsp/Thread/Main.hs#L30-L31

Example in trading bot:
https://github.com/21it/src/blob/4a4e148ae21da288d70e330527a3a383c5989843/reckless-trading-bot/src/RecklessTradingBot/Thread/Main.hs#L22-L26